src/Adapter/Product/Combination/CommandHandler/UpdateCombinationFromListingHandler.php

```diff
@@ -0,0 +1,113 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Combination\CommandHandler;
+
+use Combination;
+use PrestaShop\PrestaShop\Adapter\Product\Combination\Repository\CombinationRepository;
+use PrestaShop\PrestaShop\Adapter\Product\Combination\Update\CombinationStockUpdater;
+use PrestaShop\PrestaShop\Adapter\Product\Combination\Update\DefaultCombinationUpdater;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Command\UpdateCombinationFromListingCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\CommandHandler\UpdateCombinationFromListingHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Exception\CannotUpdateCombinationException;
+
+/**
+ * Handles @see UpdateCombinationFromListingCommand using legacy object model
+ */
+final class UpdateCombinationFromListingHandler implements UpdateCombinationFromListingHandlerInterface
+{
+    /**
+     * @var CombinationRepository
+     */
+    private $combinationRepository;
+
+    /**
+     * @var DefaultCombinationUpdater
+     */
+    private $defaultCombinationUpdater;
+
+    /**
+     * @var CombinationStockUpdater
+     */
+    private $combinationStockUpdater;
+
+    /**
+     * @param CombinationRepository $combinationRepository
+     * @param DefaultCombinationUpdater $defaultCombinationUpdater
+     * @param CombinationStockUpdater $combinationStockUpdater
+     */
+    public function __construct(
+        CombinationRepository $combinationRepository,
+        DefaultCombinationUpdater $defaultCombinationUpdater,
+        CombinationStockUpdater $combinationStockUpdater
+    ) {
+        $this->combinationRepository = $combinationRepository;
+        $this->defaultCombinationUpdater = $defaultCombinationUpdater;
+        $this->combinationStockUpdater = $combinationStockUpdater;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(UpdateCombinationFromListingCommand $command): void
+    {
+        $combination = $this->combinationRepository->get($command->getCombinationId());
+        $updatableProperties = $this->fillUpdatableProperties($combination, $command);
+        $this->combinationRepository->partialUpdate($combination, $updatableProperties, CannotUpdateCombinationException::FAILED_UPDATE_LISTED_COMBINATION);
+
+        $this->combinationStockUpdater->update($combination, $updatableProperties);
+
+        if (true === $command->isDefault()) {
```

Comment:
Yoda condition? I love it ❤️  (but I know some other people who dont :trollface: )